File: src/test/java/com/github/kayjamlang/tests/expressions/CastExpressionTest.java
package com.github.kayjamlang.tests.expressions;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.Type;
import com.github.kayjamlang.core.expressions.CastExpression;
import com.github.kayjamlang.core.expressions.ValueExpression;
import com.github.kayjamlang.tests.TestsUtils;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class CastExpressionTest {

    @Test
    public void test() throws Exception {
        Expression expression = TestsUtils.parse("123 as int");
        assertNotNull(expression);
        assertEquals(CastExpression.class, expression.getClass());

        CastExpression castExpression = (CastExpression) expression;
        assertEquals(ValueExpression.class, castExpression.expression.getClass());
        assertEquals(Type.INTEGER, castExpression.castType);
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/CastExpressionTest.scala
package com.github.kayjamlang.tests.expressions

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser, Type}
import com.github.kayjamlang.core.expressions.{CastExpression, ValueExpression}
import org.junit.Assert._
import org.junit.{BeforeClass, Test}

object CastExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("123 as int"))
  }
}

class CastExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = CastExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertEquals(classOf[CastExpression], expression getClass)
    val castExpression = expression.asInstanceOf[CastExpression]
    assertEquals(classOf[ValueExpression], castExpression.expression getClass)
    assertEquals(Type INTEGER, castExpression.castType)
  }
}
